File: backend/bu_service/app/database/models/bu_model.py
from typing import Optional

from pydantic import BaseModel

from app.database.models.merkle_tree_info_model import MerkleTreeInfo


class BuModel(BaseModel):
    eleicoes: list[int]
    filename: str
    UF: str
    zona: int
    secao: int
    municipio: str
    bu_json: str
    bu: bytes
    merkletree_info: Optional[dict[str, MerkleTreeInfo]] = {}

File: backend/public/src/database/models/bu.model.js
const mongoose = require("mongoose");

const merkletree_info = new mongoose.Schema({
    tree_name: String,
    index: Number,
    hash: String,
});

const schema_bu = new mongoose.Schema({
    eleicoes: [Number],
    filename: String,
    UF: String,
    zona: Number,
    secao: Number,
    municipio: String,
    bu_json: String,
    bu: Buffer,
    merkletree_info: { type: Map, of: merkletree_info }
});

schema_bu.index({ UF:1 });
schema_bu.index({ zona:1 });
schema_bu.index({ secao:1 });
schema_bu.index({ municipio:1 });
schema_bu.index({ eleicoes:1 });
schema_bu.index({ merkletree_info:1 });
schema_bu.index({ UF:1, zona:1, secao:1 });

module.exports = { schema_bu }